akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -186,7 +186,7 @@ private[akka] object FanIn {
     }
 
     def inputsAvailableFor(id: Int) = new TransferState {
-      override def isCompleted: Boolean = depleted(id)
+      override def isCompleted: Boolean = depleted(id) || cancelled(id)
```

Comment:
hm, but it is the downstream that is cancelled. Isn't this the wrong place to handle this?